clash-prelude/src/Clash/Class/Resize.hs

```diff
@@ -37,3 +47,65 @@ class Resize (f :: Nat -> Type) where
   signExtend = resize
   -- | Remove bits from the MSB
   truncateB :: KnownNat a => f (a + b) -> f a
+
+-- | Helper function of 'fromIntegralChecked' and 'resizeChecked'
```

Comment:
I presume you mean `checkedFromIntegral` and `checkedResize`?

Reply:
I do!